### PR TITLE
Search User Location in Landing Page

### PR DESCRIPTION
**Pull request by bon-carpo**
# To DO's

- [x] Only ask for the location of the user when the user actually clicks "Your Current Location"
- [x] Show the dropdown when the user clicks the search input field (i.e., the search input field is on focus)
- [x] Allow up and down using arrow key in dropdown selection
- [x] Fix UI Navigation
- [ ] Display Current location inside search input in KYH page when "Your Current Location" is selected and fly to kyh

# Screenshots

## Search Input on Focus
image

## Navigation Old
image

## Navigation New
image

### CODE REVIEW

**Comment by kennethliporada** [comment]
# Description
- reverse geocode user coordinates from localstorage
-  stored reverse geocode value to localstorage
- changing placeholder from user location to kyh page search
- changing placeholder from selected place to kyh page search

# Loom
- https://www.loom.com/share/f9ea2e9142ef40669e9e1f56106c05dd

# Encounter Error inside KYH page
- https://www.loom.com/share/27302802c4f8457ca26909004fc71d50

# Errors
- inside kyh page di napapalitan yung placeholder kapag na select na yung user location
- from landing page kapag nag select ng place at napunta sa kyh page then na back nandun pa rin yung placeholder na naselect mo.

**Comment by kennethliporada** [comment]
# Description
Fixed Errors:
- `inside kyh page di napapalitan yung placeholder kapag na select na yung user location`
-  `from landing page kapag nag select ng place at napunta sa kyh page then na back nandun pa rin yung placeholder na naselect mo`.

# Loom
https://www.loom.com/share/d1f9b027aad24d13853e22348757d261

# Error
- inside kyh kapag nag select ng other place then nag click ng `Your Location` hindi nag ccenter

**Review by jadurani — CHANGES_REQUESTED** [review]

File: src/app/features/know-your-hazards/services/kyh.service.ts [hunk 1]
```diff
@@ -20,6 +20,15 @@ export class KyhService {
     private hazardsService: HazardsService
   ) {}
 
+  keyBoard: Subject<any> = new Subject<any>();
```
[the remark this answers is not in the document]
Comment (kennethliporada) [reply src/app/features/know-your-hazards/services/kyh.service.ts:23]: to read arrow keys po yan.

**Comment by kennethliporada** [comment]
# Description
- getting userlocation with reverse geocode
- display placeholder to kyh page
- move arrow keys up, down and select using enter key
- stored user coordinates and user place name to the localstorage
- deleted unused variables and console logs

# Loom 
https://www.loom.com/share/b96ad5351ad64d0bbdec4140873f7881

**Comment by jadurani** [comment]
reworked at https://github.com/UPRI-NOAH/noah-frontend/pulls

**Pull request closed without merging**